Zend/Optimizer/block_pass.c

```diff
@@ -470,7 +470,58 @@ static void zend_optimize_block(zend_basic_block *block, zend_op_array *op_array
 					goto optimize_bool;
 				}
 				break;
+			case ZEND_IS_IDENTICAL:
+				if (opline->op1_type == IS_CONST &&
+					opline->op2_type == IS_CONST) {
+					goto optimize_constant_binary_op;
+				}
 
+				if (opline->op1_type == IS_CONST &&
+					(Z_TYPE(ZEND_OP1_LITERAL(opline)) <= IS_TRUE && Z_TYPE(ZEND_OP1_LITERAL(opline)) >= IS_NULL)) {
+					opline->opcode = ZEND_TYPE_CHECK;
+					opline->extended_value = (1 << Z_TYPE(ZEND_OP1_LITERAL(opline)));
+					COPY_NODE(opline->op1, opline->op2);
+					SET_UNUSED(opline->op2);
+					++(*opt_count);
+					goto optimize_type_check;
+				} else if (opline->op2_type == IS_CONST &&
+					(Z_TYPE(ZEND_OP2_LITERAL(opline)) <= IS_TRUE && Z_TYPE(ZEND_OP2_LITERAL(opline)) >= IS_NULL)) {
+					opline->opcode = ZEND_TYPE_CHECK;
+					opline->extended_value = (1 << Z_TYPE(ZEND_OP2_LITERAL(opline)));
+					SET_UNUSED(opline->op2);
+					++(*opt_count);
+					goto optimize_type_check;
+				}
+				break;
+			case ZEND_TYPE_CHECK:
+				optimize_type_check:
```

Comment:
Nit: We almost universally put labels on column 0.

Reply:
I could swear that block_pass.c was an exception to this, but it appears it is not. Not sure what happened.

Reply:
Ah, it was `optimize_bool` that already was indented.